### PR TITLE
New: Health Check Failure Notifications

### LINKED ISSUE #295: Add notifications onHealthCheck Issue
### Description of request and what problem are you looking to solve?

Send a notification out when a new health error occurs (indexer failures, etc.)

https://github.com/Sonarr/Sonarr/issues/1798

### PR DESCRIPTION
**Pull request by Qstick**
#### Database Migration
YES (No 29 - Add Columns to notification table)

#### Description
Adds ability to get notifications on health check failures from Lidarr. 

- Only alerts on new instances and not every time health check runs. (avoids spam)
- Ability for filter Errors and Warnings, or Errors only
- Adds Support to most notification types (Including custom script)
- Adds OnDownloadFailure Notification
- Adds OnImportFailure Notification (Needs future expansion)
- Adds OnTrackRetag Notification

#### Todos
- [ ] Documentation (Wiki Pages)

#### Screenshots
image
image

#### Issues Fixed or Closed by this PR
Fixes #295
*

### CODE REVIEW

**Comment by Qstick** [comment]
Note: Should we go ahead and add OnFailedImport to notification table in this DB Migration. This will be possible after #592 is merged.

**Comment by Qstick** [comment]
We need to bump migration number on this as well

**Comment by ta264** [comment]
I've rebased and added tag rename notifications.  Can merge the migrations if that's better.

**Comment by Qstick** [comment]
Yeah let's merge the migration to one

**Comment by ta264** [comment]
I've merged the migration and pulled in the Discord notification from Sonarr

**Comment by Qstick** [comment]
@ta264 This looks good to me and seems to work fine in testing. Its pretty straight-forward. We should likely look at some way to better show all the options in the UI at some point in future. The list in the modal doesn't really segregate out the notification types from the other options so well. 

You'll have to hit approve on this since I opened the PR.

**Comment by Qstick** [comment]
Also, wouldn't mind blitzing out `On Track Import` in this migration. I really don't think its that useful. May make more sense to get rid of it, and rename On Album Import to On Release Import. We could pass all the tracks imported in that single event.  Thoughts?

File: src/NzbDrone.Core/Notifications/Plex/HomeTheater/PlexHomeTheater.cs [hunk 1]
```diff
@@ -26,11 +26,6 @@ public override void OnGrab(GrabMessage grabMessage)
             Notify(ALBUM_GRABBED_TITLE_BRANDED, grabMessage.Message);
         }
 
-        public override void OnDownload(TrackDownloadMessage message)
-        {
-            Notify(TRACK_DOWNLOADED_TITLE_BRANDED, message.Message);
-        }
-
```
Comment (Qstick) [inline src/NzbDrone.Core/Notifications/Plex/HomeTheater/PlexHomeTheater.cs]: Add Notify of ReleaseImport

File: src/NzbDrone.Core/Notifications/Webhook/Webhook.cs [hunk 2]
```diff
@@ -41,29 +41,6 @@ public override void OnGrab(GrabMessage message)
...
-                    QualityVersion = trackFile.Quality.Revision.Version,
-                    ReleaseGroup = trackFile.ReleaseGroup
-                }),
-                TrackFile = new WebhookTrackFile(trackFile),
-                IsUpgrade = message.OldFiles.Any()
-            };
-
-            _proxy.SendWebhook(payload, Settings);
-        }
-
```
Comment (Qstick) [inline src/NzbDrone.Core/Notifications/Webhook/Webhook.cs]: Add OnReleaseImport Handler?

File: frontend/src/Store/Actions/Settings/notifications.js [hunk 3]
```diff
@@ -103,7 +103,6 @@ export default {
     [SELECT_NOTIFICATION_SCHEMA]: (state, { payload }) => {
       return selectProviderSchema(state, section, payload, (selectedSchema) => {
         selectedSchema.onGrab = selectedSchema.supportsOnGrab;
-        selectedSchema.onDownload = selectedSchema.supportsOnDownload;
```
Comment (Qstick) [inline frontend/src/Store/Actions/Settings/notifications.js]: Do we need to add all the new notification triggers here?

**Pull request merged**